### ISSUE TITLE
--strict-equality should work with literal types

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#6672]
This example should give an error with `--strict-equality`:
```python
def returns_a_or_b() -> Literal['a', 'b']:
  ...

if returns_a_or_b() == 'c':  # This can never by True
    ...
```
this depends on the implementation of type narrowing by equality checks, see https://github.com/python/mypy/issues/5935. Basically, `Literal[...].__eq__()` should provide a literal context for its argument.
